sqldelight-gradle-plugin/src/main/kotlin/app/cash/sqldelight/gradle/SqlDelightExtension.kt

```diff
@@ -1,15 +1,17 @@
 package app.cash.sqldelight.gradle
 
 import groovy.lang.Closure
+import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.Project
+import org.gradle.api.provider.Property
 import org.gradle.util.ConfigureUtil
 
-open class SqlDelightExtension {
-  internal val databases = mutableListOf<SqlDelightDatabase>()
+abstract class SqlDelightExtension {
+  internal abstract val databases: NamedDomainObjectContainer<SqlDelightDatabase>
   internal var configuringDatabase: SqlDelightDatabase? = null
   internal lateinit var project: Project
 
-  var linkSqlite = true
+  abstract val linkSqlite: Property<Boolean>
 
   fun methodMissing(name: String, args: Any): Any {
```

Comment:
Do we still need this method for Groovy users? I think, Gradle named API take care of it